site/content/docs/user/rootless.md

```diff
@@ -9,57 +9,214 @@ menu:
-  performance.)
+4. Next, restart the `systemd-modules-load` service to make these changes effective immediately:
 
-  Please note that:
+   ```sh
+   sudo systemctl restart systemd-modules-load.service
+   ```
 
-  - `/etc/systemd/system/user@.service.d/` directory needs to be created if not already present on your host
-  - If using Docker and it was already running when this step was done, a restart is needed for the changes to take
-    effect
-      {{< codeFromInline lang="bash" >}}
-      systemctl --user restart docker
-      {{< /codeFromInline >}}
+5. Alternatively, restart your system to ensure these changes take effect.
 
-- Create `/etc/modules-load.d/iptables.conf` with the following content:
+### Increase PID Limits
 
-  ```
-  ip6_tables
-  ip6table_nat
-  ip_tables
-  iptable_nat
-  ```
+KIND nodes are represented as individual containers on their hosts. Runtimes such as podman set
+default [process id limits](https://docs.podman.io/en/v4.3/markdown/options/pids-limit.html#pids-limit-limit)
+that may be too low for the node or for a pod running on the node. The Ingress NGINX Controller is
+[particularly susceptible](https://github.com/kubernetes-sigs/kind/issues/3451) to this issue.
 
-- If using podman, be aware that by default there is a [limit](https://docs.podman.io/en/v4.3/markdown/options/pids-limit.html#pids-limit-limit) to the number of pids that can be created. This can cause problems like nginx workers inside a container not spawning correctly.
-    - If you want to disable this limit, edit your `containers.conf` file (generally located in `/etc/containers/containers.conf`). Note that this could cause things like pid exhaustion to happen on the host machine. Alternatively, change `0` to your desired new limit:
+To increase the PID limit, do the following:
+
+1. If using podman, edit your `containers.conf` file (generally located in
+   `/etc/containers/containers.conf` or `~/.config/containers/containers.conf`) to increase the PIDs
+   limit to a desired value (default 4096 on most systems):
 
     ```ini
     [containers]
-    pids_limit = 0
+    pids_limit = 65536
     ```
 
+2. Re-recreate the KIND cluster for these changes to take effect:
+
+   ```sh
+   kind delete cluster && kind create cluster
+   ```
+
+### Increase inotify Limits
+
+As documented in [known issues](/docs/user/known-issues/#pod-errors-due-to-too-many-open-files), pods may
+fail by reaching inotify watch and instance limits. Ingress controllers such as NGINX and Contour
+are particularly susceptible to this issue.
+
+To increase the inotify limits, do the following:
+
+1. As root, create a `.conf` file in `/etc/systctl.d` that increases the `fs.inotify` max user settings:
+
+   ```
+   fs.inotify.max_user_watches = 524288
+   fs.inotify.max_user_instances = 512
+   ```
+
+2. Reload `sysctl` for these changes to take effect:
+
+   ```sh
+   sudo sysctl --system
+   ```
+
+Alternatively, restart your system for these changes to take effect.
+
```

Comment:
nit: If we're going to be picky about having a blank line after a header, we should also be picky about not having multiple blank lines. ;)